include/amici/solver.h

```diff
@@ -100,11 +100,14 @@ class Solver {
      * @param dx0 initial derivative states
      * @param sx0 initial state sensitivities
      * @param sdx0 initial derivative state sensitivities
+     * @param steadystate (optional) flag indicating steadystate case
+     * @param xQ0 initial quadrature vector
      */
 
     void setup(realtype t0, Model *model, const AmiVector &x0,
                const AmiVector &dx0, const AmiVectorArray &sx0,
-               const AmiVectorArray &sdx0) const;
+               const AmiVectorArray &sdx0, bool steadystate = false,
+               const AmiVector &xQ0 = AmiVector(0)) const;
```

Comment:
Somehow not happy to modify the setup routine here... But need it to inform the solver, that we want to do forward quadratures and use another RHS function... Any ideas on that? Could this be done better?

Reply:
I see your point. Is is possible to write a function that creates a solver instance using the regular setup routine and then just attaches different functions where appropriate by calling all the init functions again?

Reply:
This would have the added benefit that the logic why these things are set the way they are can be documented in a central location.

Reply:
Seems not to be possible naively, as re-calling CVode/IDAInit leads to unwanted behavior (it unsets the linear solver module, for example).
Trying to set it in the solver memory directly now.